pogobuf/pogobuf.utils.js

```diff
@@ -10,37 +10,36 @@ var s2 = require('s2-geometry').S2;
      * @param {number} lat
      * @param {number} lng
+     * @param {number} radius
      * @returns {array}
      * @static
      */
-    getCellIDs: function(lat, lng) {
+    getCellIDs: function(lat, lng, radius) {
+        if(typeof radius === 'undefined') radius = 3;
+        
         var origin = s2.S2Cell.FromLatLng({ lat: lat, lng: lng }, 15);
         var cells = [];
-
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] - 2, origin.ij[1] - 2], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] - 2, origin.ij[1] - 1], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] - 1, origin.ij[1] - 1], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] - 1, origin.ij[1] - 2], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0], origin.ij[1] - 2], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] + 1, origin.ij[1] - 2], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] + 1, origin.ij[1] - 1], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0], origin.ij[1] - 1], origin.level).toHilbertQuadkey());
-        cells.push(origin.toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] + 1, origin.ij[1]], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] + 1, origin.ij[1] + 1], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0], origin.ij[1] + 1], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] - 1, origin.ij[1] + 1], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] - 1, origin.ij[1]], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] - 2, origin.ij[1]], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] - 2, origin.ij[1] + 1], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] - 2, origin.ij[1] + 2], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] - 1, origin.ij[1] + 2], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0], origin.ij[1] + 2], origin.level).toHilbertQuadkey());
-        cells.push(s2.S2Cell.FromFaceIJ(origin.face, [origin.ij[0] + 1, origin.ij[1] + 2], origin.level).toHilbertQuadkey());
-
+         
+        cells.push(origin.toHilbertQuadkey()); // middle block
+         
+        for(var i = 1; i < radius; i++) {
```

Comment:
For those of us who don't understand how the S2 algorithm works, could you provide just a brief (single sentence or two) overview of what you're doing, and why? I see a lot of libraries that take the S2 stuff as 'magic', and then have poor results. It'd be nice if this was just a little better documented so someone unfamiliar could at least follow along a little easier.